Sources/SwiftProcessing/Core/UIKitElements/Capture.swift

```diff
@@ -0,0 +1,310 @@
+        self.cameraPosition = cameraPosition
+        checkCameraAccess { allowed in
+            if allowed { self.setup() }
+            completion?(allowed)
+            self.preparingCompletionHandler = nil
+        }
+    }
+    
+    private func setup() {
+        configureCaptureSession()
+    }
+    
+    func start() {
+        if cameraIsReadyToUse { session.startRunning() }
+    }
+    
+    func stop() {
+        session.stopRunning()
+    }
+    
+}
+
+extension Camera {
+    
+    private func askUserForCameraPermission(_ completion:  ((Bool) -> Void)?) {
+        AVCaptureDevice.requestAccess(for: AVMediaType.video) { (allowedAccess) -> Void in
+            DispatchQueue.main.async { completion?(allowedAccess) }
+        }
+    }
+    
+    private func checkCameraAccess(completion: ((Bool) -> Void)?) {
+        askUserForCameraPermission { [weak self] allowed in
+            guard let self = self, let completion = completion else { return }
+            self.cameraIsReadyToUse = allowed
+            if allowed {
+                completion(true)
+            } else {
+                self.showDisabledCameraAlert(completion: completion)
+            }
+        }
+    }
+    
+    private func configureCaptureSession() {
+        guard let previewView = previewView else { return }
+        // Define the capture device we want to use
+        
+        guard let camera = AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: cameraPosition) else {
+            let error = NSError(domain: "", code: 0, userInfo: [NSLocalizedDescriptionKey : "No front camera available"])
+            show(error: error)
+            return
+        }
+        
+        // Connect the camera to the capture session input
+        do {
+            
+            try camera.lockForConfiguration()
+            defer { camera.unlockForConfiguration() }
+            
+            if camera.isFocusModeSupported(.continuousAutoFocus) {
+                camera.focusMode = .continuousAutoFocus
+            }
+            
+            if camera.isExposureModeSupported(.continuousAutoExposure) {
+                camera.exposureMode = .continuousAutoExposure
+            }
+            
+            let cameraInput = try AVCaptureDeviceInput(device: camera)
+            session.addInput(cameraInput)
+            
+        } catch {
+            show(error: error as NSError)
+            return
+        }
+        
+        // Create the video data output
+        let videoOutput = AVCaptureVideoDataOutput()
+        videoOutput.setSampleBufferDelegate(self, queue: dataOutputQueue)
+        videoOutput.videoSettings = [kCVPixelBufferPixelFormatTypeKey as String: kCVPixelFormatType_32BGRA]
+        
+        // Add the video output to the capture session
+        session.addOutput(videoOutput)
+        
+        let videoConnection = videoOutput.connection(with: .video)
+        videoConnection?.videoOrientation = .portrait
```

Comment:
I'm thinking it makes sense to make this the devices current orientation. You can read that using 

UIDevice.current.orientation.isLandscape